packages/tasit-action/src/TasitAction.js

```diff
@@ -19,117 +22,370 @@ class Utils {
 }
 
 class Subscription {
+  #emitter;
+  #events = [];
+
+  constructor(eventEmitter) {
+    this.#emitter = eventEmitter;
+  }
+
+  removeListener = (eventName, listener) => {
```

Comment:
Yes

Reply:
Up top, I'll note that the spec for these features was far from complete (somewhat a bad thing that's might fault, but also somewhat by design to allow for creativity and collaboration on it), so this implementation is a _very_ reasonable interpretation of the spec and a great choice to make. So it's my fault if any work is duplicated because of an impartial spec / expectation mismatch here. That said, here are my thoughts on how I picture this working instead:

I think I'm leaning towards the stance that we should enforce just one listener per event.

The idea is that the user could dispatch a redux (or similar) action in an app like sending a transaction that would add a listener, and then there would be a single handler on the data from that event that changed the state of the app, and then all the appropriate presentational components would rerender as a result of that. I think this is a better pattern than each component deciding how to change their internal state with their different listeners. Even without redux, it would make sense to create a container component higher up in the app that passed this data to multiple presentational components.

Thoughts? I'm curious to hear how you picture multiple listeners being used.

Reply:
No worries, there are a lot of open threads here! That all makes sense. Okay, great! We can split this out into a dedicated issue if we want.

Reply:
https://github.com/tasitlabs/TasitSDK/issues/100

Reply:
I'm resolving this based on that PR you linked to